File: HW5/151044058_CSE241_HW5/Polygon.cpp
#include "Circle.h"
#include "Triangle.h"
#include "Rectangle.h"
#include "Polygon.h"
#include "Shape.h"

namespace HW5_151044058
{
	Polygon :: Point2D :: Point2D()
	{
		//Empty
	}
	Polygon :: Point2D :: Point2D(double x, double y)
	{
		koor_x = x;
		koor_y = y;
	}
	void Polygon :: Point2D :: setKoor_x(double koor_xValue) //set koor_x
	{
		koor_x = koor_xValue;
	}

	void Polygon :: Point2D :: setKoor_y(double koor_yValue)	//set koor_y
	{
		koor_y = koor_yValue;
	}
	double Polygon :: Point2D :: getKoor_x()const // get koor_x
	{
		return koor_x;
	}
	double Polygon :: Point2D :: getKoor_y()const // get koor_y
	{
		return koor_y;
	}
}

File: HW7/151044058_CSE241_HW7/Polygon.java
import javax.swing.JPanel;

/**
 * 
 * @author Ali Haydar Kurban
 *
 */
public abstract class Polygon extends JPanel implements Shape 
{
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	/**
	 * 
	 * @author Ali Haydar Kurban 
	 * 	
	 */
	public class Point2D
	{
		public Point2D()
		{
			//Empty
		}
		
		public Point2D(double x, double y)
		{
			koor_x = x;
			koor_y = y;
		}
		public void setKoor_x(double koor_xValue)
		{
			koor_x = koor_xValue;
		}
		public void setKoor_y(double koor_yValue)
		{
			koor_y = koor_yValue;
		}
		public double getKoor_x()
		{
			return koor_x;
		}
		public double getKoor_y()
		{
			return koor_y;
		}
		
		protected double koor_x;
		protected double koor_y;
	}

}